__tests__/app.test.js

```diff
@@ -188,4 +188,70 @@ describe("GET /api", () => {
         });
     });
   });
+
+  describe("POST /api/articles/:article_id/comments", () => {
+    test("201: Should respond with the newly posted comment", () => {
+      const newComment = {
+        username: "butter_bridge",
+        body: "This is the test comment",
+      };
+      return request(app)
+        .post("/api/articles/1/comments")
+        .send(newComment)
+        .expect(201)
+        .then(({ body: { comment } }) => {
+          expect(comment).toMatchObject({
+            comment_id: expect.any(Number),
+            body: newComment.body,
+            article_id: 1,
+            author: newComment.username,
+            votes: 0,
+            created_at: expect.any(String),
+          });
+        });
+    });
+
+    test("400: Should respond with bad request if body is not formatted correctly", () => {
+      const newComment = {
+        username: "butter_bridge",
+      };
+      return request(app)
+        .post("/api/articles/1/comments")
+        .send(newComment)
+        .expect(400)
+        .then(({ body: { msg } }) => {
+          expect(msg).toBe("Bad request");
+        });
+    });
+
+    test("404: Should return not found if article ID cannot be found", () => {
+      // unsure if this should be 404 or 400
+      const newComment = {
+        username: "butter_bridge",
+        body: "This is the test comment",
+      };
+      return request(app)
+        .post("/api/articles/1000/comments")
+        .send(newComment)
+        .expect(404)
+        .then(({ body: { msg } }) => {
+          expect(msg).toBe("Article not found");
+        });
+    });
+
+    test("404: Should return not found if username is not found", () => {
+      // unsure if this should be 404 or 400
+      const newComment = {
+        username: "invalid_username",
+        body: "This is the test comment",
+      };
+      return request(app)
+        .post("/api/articles/1/comments")
+        .send(newComment)
+        .expect(404)
+        .then(({ body: { msg } }) => {
+          expect(msg).toBe("User not found");
+        });
+    });
+  });
```

Comment:
Testing looks good, could also add:
- Status 400, invalid ID, e.g. string of "not-an-id"
- 404s for non-existent user / article looks good to me